xun/tests/test_functions.py

```diff
@@ -306,24 +305,64 @@ def h():
     assert result == 'abbc'
 
 
-@pytest.mark.skip(reason="Structured unpacking is not supported yet")
 def test_structured_unpacking_list():
     @xun.function()
     def f():
-        return ('a', 'b')
+        return ('a', ('b', 'c'))
 
     @xun.function()
     def h():
         with ...:
-            [a, b] = f()
-        return a + b
+            [a, [b, c]] = f()
+        return a + b + c
 
     result = h.blueprint().run(
         driver=xun.functions.driver.Sequential(),
         store=xun.functions.store.Memory(),
     )
 
-    assert result == 'ab'
+    assert result == 'abc'
+
+
+def test_structured_unpacking_starred():
+    @xun.function()
+    def f():
+        return ('a', 'b', 'c', 'd')
+
+    @xun.function()
+    def h():
+        with ...:
+            a, *bc, d = f()
+        b, c = bc
+        return a + b + c + d
+
+    result = h.blueprint().run(
+        driver=xun.functions.driver.Sequential(),
+        store=xun.functions.store.Memory(),
+    )
```

Comment:
Idea: 
```suggestion
    result = h.blueprint().run(driver=TestDriver, store=MemoryStore())
```

Reply:
Good idea, I will create a new issue out of this